Digital_Fusion/src/components/CardContainer/CardContainer.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { Children, useState } from "react";
+import { cardData } from "../../utils/constants";
+import Popup from "../PopupForm/Popup";
+import { Card } from "../Card/Card";
+export function CardContainer() {
+  const [isOpen, setIsOpen] = useState(false);
+  const [modalContent, setModalContent] = useState("");
+
+  const openPopup = (card) => {
+    setModalContent(card);
+    setIsOpen(true);
+  };
+
+  const closePopup = () => {
+    setIsOpen(false);
+    setModalContent("");
+  };
+
+  return (
+    <main>
+      <div className="card__container">
+        {cardData.map((card) => (
+          <Card key={card.id} card={card} openPopup={openPopup}></Card>
```

Comment:
Good job that you've used `id` for the `key` rather than the `index`.

Here is why we shouldn’t use `index` for that: [click here]( https://programmingwithmosh.com/react/why-do-i-need-keys-in-react-lists/#:~:text=React%20recommends%20that%20you%20do,to%20some%20unstable%20component%20behaviour.)